torch/fx/passes/fuser_utils.py

```diff
@@ -0,0 +1,228 @@
+from queue import SimpleQueue
+from typing import List, Optional, Dict
+
+import torch.fx
+from torch.fx.graph import map_arg
+from torch.fx.passes.tools_common import NodeList, NodeSet, legalize_graph
+from torch.fx.passes.split_utils import HolderModule
+
+def topo_sort(nodes: NodeList) -> NodeList:
+    # sort nodes according to the topological order
+    indegree_map = {node : 0 for node in nodes}
+    candidates = SimpleQueue()
+
+    for node in nodes:
+        for n in node.all_input_nodes:
+            if n in indegree_map:
+                indegree_map[node] += 1
+        if indegree_map[node] == 0:
+            candidates.put(node)
+
+    sorted_nodes: NodeList = list()
+    while not candidates.empty():
+        node = candidates.get()
+        sorted_nodes.append(node)
+
+        for n in node.users:
+            if n in indegree_map:
+                indegree_map[n] -= 1
+                if indegree_map[n] == 0:
+                    candidates.put(n)
+
+    return sorted_nodes
```

Comment:
I'm surprised that we do not have a `topo_sort` somewhere.

Nitpick: can we assert `len(sorted_nodes) == len(nodes)`?

Reply:
yes, added.